### PR TITLE
Add source pause feature

### PR DESCRIPTION
**Pull request by rvsia**
https://issues.redhat.com/browse/RHCLOUD-14217

TODO
- [x] backend https://github.com/RedHatInsights/sources-api/pull/402
  - [x] update all tests where source is being pause/resumed
  - [x] tests
- [x] order of buttons in action resolver
- [x] add pause icon to source status
- [x] notifications https://issues.redhat.com/browse/RHCLOUD-14390
- [x] change title of Edit credentials form

**After**

Screenshot 2021-05-25 at 16 56 12
Screenshot 2021-05-25 at 16 56 02
Screenshot 2021-05-25 at 16 55 59
Screenshot 2021-05-25 at 16 55 55
Screenshot 2021-05-25 at 16 59 32
Screenshot 2021-05-25 at 16 55 08
Screenshot 2021-05-25 at 16 58 03
Screenshot 2021-05-25 at 16 55 02
Screenshot 2021-05-25 at 16 56 58
Screenshot 2021-05-25 at 16 56 56
Screenshot 2021-05-25 at 16 56 48
Screenshot 2021-05-25 at 16 56 18

### CODE REVIEW

File: src/components/CredentialsForm/CredentialsForm.js [hunk 1]
```diff
@@ -80,7 +86,9 @@ const CredentialsForm = () => {
     <SourcesFormRenderer
       clearedValue={null}
       schema={{
-        fields: [generateSuperKeyFields(sourceTypes, sourceTypeName)],
+        fields: [...generateSuperKeyFields(sourceTypes, sourceTypeName)].map((field) =>
```
Comment (Hyperkid123) [inline src/components/CredentialsForm/CredentialsForm.js:89]: Is it necessary to duplicate the output array?

File: src/redux/sources/reducer.js [hunk 2]
```diff
@@ -39,7 +39,7 @@ export const entitiesPending = (state, { options }) => ({
 export const entitiesLoaded = (state, { payload: rows, options }) => ({
   ...state,
   loaded: Math.max(state.loaded - 1, 0),
-  entities: rows,
+  entities: rows, //.map((row) => ({ ...row, paused_at: 'today' })),
```
Comment (Hyperkid123) [inline src/redux/sources/reducer.js:42]: Is this map function supposed to be executed or it just some left over?

File: src/test/api/entities.test.js [hunk 3]
```diff
@@ -249,6 +249,24 @@ describe('entities spec', () => {
 
         expect(result).toEqual(OK_RESPONSE);
       });
+
+      it('pauseSource', async () => {
+        const method = 'Post';
+        mock[`on${method}`](`/api/sources/v3.1/sources/${SOURCE_ID}/pause`).reply(200, OK_RESPONSE);
```
Comment (Hyperkid123) [inline src/test/api/entities.test.js:255]: Are you sure you want to use mock for async calls?

Comment (rvsia) [reply src/test/api/entities.test.js:255]: This is a test of the url endpoint.

**Review by Hyperkid123 — APPROVED** [review]

**Pull request merged**